api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+
+  try {
+    const accessToken = req.get('X-Access-Token');
+    if (accessToken === undefined) {
+      res.status(401).json('authentication required');
+    }
+    req.user = jwt.verify(accessToken, process.env.TOKEN_SECRET);
+    next();
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
If we catch an error we always want to pass it to `next()` so that our error middleware can deal with it and our client at least receives a response.